### PR TITLE
Create an Eslint reporter

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-frontend/issues/2281

---

I tried to use an existing lib but it was super slow and I hate when things are super slow. I wrote another reporter, taking inspiration from the excellent Flow reporter and I added a nice trick to make Eslint report blazing fast ⚡️ 

The trick is to leverage `git` to find the modified files (newly created or modified, not deleted) and to pass this list to ESLint.

## Screenshots

screen shot 2018-11-13 at 18 55 27

screen shot 2018-11-13 at 19 00 32

### CODE REVIEW

**Comment by willdurand** [comment]
I could probably improve the detection of a file that has been both renamed and modified:

```
RM src/amo/sagas/categories.js -> src/core/sagas/categories.js
```

**Comment by willdurand** [comment]
> I could probably improve the detection of a file that has been both renamed and modified:
> 
> ```
> RM src/amo/sagas/categories.js -> src/core/sagas/categories.js
> ```

done.

**Comment by willdurand** [comment]
I played with this patch for some time now,and it works as I hoped.

**Comment by kumar303** [comment]
Sorry, I didn't get a chance to play with this before heading out on PTO week. Feel free to get feedback from someone else and land it 🌵

**Comment by willdurand** [comment]
I'd really like to have this reviewed/merged

**Comment by bobsilverberg** [comment]
> I'd really like to have this reviewed/merged.

Ok, it'll be next on my list. I can only review one of your PRs at a time, you know! 😉

**Comment by willdurand** [comment]
> Also, when I run `yarn test` locally I get an error because of a problem with the git command. I'm guessing I must be on an older version of git than you. I'm on 2.7.1. Which version are you on?

ah, I suppose the option with value came later. I can remove the value I think.

File: tests/jest-reporters/utils.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    .map((line) => {
+      // Return the new name of a renamed file.
+      if (line.startsWith('RM')) {
+        return line.substring(line.indexOf('->'));
+      }
+
+      return line;
+    })
+    .map((line) => line.split(' '))
+    // .flatMap()
```
[the remark this answers is not in the document]
Comment (willdurand) [reply tests/jest-reporters/utils.js]: ah not really, it should really be `flatMap()` but this is not implemented yet..

**Comment by willdurand** [comment]
> Updating git to the latest version fixed the error, and it all looks great!

oh cool, then no need to do anything. It feels safer to rely on the 1st version of `--porcelain`, the 2nd version is not backward compatible.

**Comment by willdurand** [comment]
> I notice there are no tests for the reporter itself. Do you think we should have any?

I think it is fine for now. The code is simple and I would not really know what to test. The Flow reporter isnt tested either, which does not mean we should not test those files but.. I guess as far as it does not break our dev envs, that's fine.

**Pull request merged**